### ISSUE TITLE
About a error in retrain a xlm model

### ISSUE THREAD

**Issue by zxk19981227** [huggingface/transformers#12382]
when i train a xlm model as following code, an error occurs that 'label' is a unexpected for forward. Anyway to solve it?
`# -*- coding: UTF-8 -*-
from transformers import XLMTokenizer, XLMModel, Trainer
from datasets import load_dataset, Dataset
from transformers import LineByLineTextDataset, TrainingArguments
from transformers.data.data_collator import DataCollatorForLanguageModeling

model = XLMModel.from_pretrained('xlm-mlm-tlm-xnli15-1024')
tokenizer = XLMTokenizer.from_pretrained('xlm-mlm-tlm-xnli15-1024')
# train_datasets=load_dataset('text',data_files={'train':'./tmp/xxx.train.txt','valitation':'./tmp/all_val_data.txt'})

# 将我们刚刚加载好的datasets ，通过tokenizer做映射，得到input_id，也就是实际输入模型的东西。
# def tokenize_function(examples):
#     # Remove empty lines
#     examples["text"] = [line for line in examples["text"] if len(line) > 0 and not line.isspace()]
#     return tokenizer(
#         examples["text"],
#         padding="max_length", # 进行填充
#         truncation=True, # 进行截断
#         max_length=256, # 设置句子的长度
#         # We use this option because DataCollatorForLanguageModeling (see below) is more efficient when it
#         # receives the `special_tokens_mask`.
#         return_special_tokens_mask=True,
#     )

# 得到训练集和验证集
# train_dataset = tokenized_datasets["train"]
# eval_dataset = tokenized_datasets["validation"]
model.resize_token_embeddings(len(tokenizer))
train_dataset = LineByLineTextDataset(tokenizer=tokenizer,
                                      file_path=''
                                      block_size=512)
datacollector = DataCollatorForLanguageModeling(tokenizer, mlm=True, mlm_probability=0.15)
# train_method=Trainer(model=model,data_collator=datacollector,train_dataset=train_dataset)
training_args = TrainingArguments(output_dir='./outputs/', overwrite_output_dir=True, num_train_epochs=20,
                                  learning_rate=6e-5,
                                  per_device_train_batch_size=128, save_total_limit=10)  # save_steps=10000
trainer = Trainer(
    model=model, args=training_args, data_collator=datacollector, train_dataset=train_dataset)

trainer.train()
trainer.save_model('./outputs/')
`
